### PR TITLE
Add UEFITool

### PR DESCRIPTION
**Pull request by vulpes2**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission]. (AppStream data will be added at a later date)
- [x] I am using only the minimal set of permissions.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I contacted upstream developers about submitting their software to Flathub. 
    - There has already been some discussions somewhat related to packaging: https://github.com/LongSoft/UEFITool/issues/282
    - Upstream had no objections on this old issue while it was open: https://github.com/LongSoft/UEFITool/issues/86
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by joshua-stone** [comment]
bot, build com.github.LongSoft.UEFITool

**Comment by vulpes2** [comment]
bot, build com.github.LongSoft.UEFITool

**Comment by vulpes2** [comment]
bot, build com.github.LongSoft.UEFITool

**Comment by hfiguiere** [comment]
has the appstream file been submitted upstream?

**Comment by vulpes2** [comment]
Not yet, the appstream file will be submitted at a later date with some other build system related changes I have in mind. I thought that was a common practice to keep downstream appstream files in flatpak repos, since a lot of repos under the flathub organization seems to have them.

**Comment by hfiguiere** [comment]
appstream files should be upstream. It is NOTHING specific to Flathub/Flatpak.

**Comment by vulpes2** [comment]
Upstream is not going to make a new release just because the appstream data has been added, and they have no fixed release schedules. Would it be acceptable for me to submit it to upstream while still keeping it downstream until the next release?

**Comment by hfiguiere** [comment]
who talked about a new release?

**Comment by vulpes2** [comment]
If I submit a PR containing the appstream data and it was merged, it won't be included in the current release, so I will still have to ship it in this repo until the next release.

**Comment by vulpes2** [comment]
The metadata has been submitted to the upstream repo.

File: com.github.LongSoft.UEFITool.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+runtime-version: "6.3"
+sdk: org.kde.Sdk
+rename-desktop-file: uefitool.desktop
+rename-icon: uefitool
+command: uefitool
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --filesystem=home
```
Comment (JakobDev) [inline com.github.LongSoft.UEFITool.yml:12]: Why is this needed?

File: com.github.LongSoft.UEFITool.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline com.github.LongSoft.UEFITool.appdata.xml:1]: Please make sure to submit this upstream.

File: com.github.LongSoft.UEFITool.yml (hunk 1 above)
Comment (barthalion) [inline com.github.LongSoft.UEFITool.yml:12]: Looks like it uses the portals perfectly fine without `--filesystem=home`. Can you double-check if everything works for you, and if yes, drop it?

File: com.github.LongSoft.UEFITool.appdata.xml (hunk 2 above)
Comment (vulpes2) [reply com.github.LongSoft.UEFITool.appdata.xml:1]: https://github.com/LongSoft/UEFITool/pull/299

File: com.github.LongSoft.UEFITool.yml (hunk 1 above)
Comment (vulpes2) [reply com.github.LongSoft.UEFITool.yml:12]: Thank you for the review, `--filesystem=home` is indeed unnecessary and the application works fine without home access.

**Comment by vulpes2** [comment]
`--share=ipc` and `--filesystem=home` have been removed after further testing

File: com.github.LongSoft.UEFITool.yml (hunk 1 above)
Comment (vulpes2) [reply com.github.LongSoft.UEFITool.yml]: `--share=ipc` and `--filesystem=home` have been removed after further testing.

**Comment by hfiguiere** [comment]
`--share=ipc` is needed for x11 (even fallback) it was never a question,

**Comment by barthalion** [comment]
bot, build com.github.LongSoft.UEFITool

**Comment by vulpes2** [comment]
@NikolajSchlej would you like write access to this repo as well?

**Comment by NikolajSchlej** [comment]
Sure, but I hope we won't break unixbuild.sh, so changes here won't be necessary.
